samza-api/src/main/java/org/apache/samza/operators/windows/WindowInternal.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.operators.windows;
```

Comment:
Actually, I tried but ended up not moving this one. The problem is the factory methods of `Windows` refer to `WindowInternal` to internally instantiate a `Window` instance. 

Currently, `WindowInternal` is declared with a `package private` constructor (so that it can't be instantiated outside its package).  Moving `WindowInternal` to its own package would mean that the constructor would have to be public. (which I 'd love to avoid).